File: basearchcomponents/src/main/java/com/tanveershafeeprottoy/basearchcomponents/viewmodels/BaseViewModel.java
package com.tanveershafeeprottoy.basearchcomponents.viewmodels;

import com.tanveershafeeprottoy.basearchcomponents.livedata.SingleLiveEvent;

import androidx.lifecycle.ViewModel;

/**
 * @author Tanveer Shafee Prottoy
 */
public abstract class BaseViewModel extends ViewModel {
    protected SingleLiveEvent<Throwable> throwableSingleLiveEvent = new SingleLiveEvent<>();

    protected void throwError(Throwable throwable) {
        throwableSingleLiveEvent.setValue(throwable);
    }
}

File: basearchcomponents/src/main/java/com/tanveershafeeprottoy/basearchcomponents/viewmodels/BaseViewModel.kt
package com.tanveershafeeprottoy.basearchcomponents.viewmodels

import androidx.lifecycle.ViewModel
import com.tanveershafeeprottoy.basearchcomponents.interfaces.BaseViewModel
import com.tanveershafeeprottoy.basearchcomponents.livedata.SingleLiveEvent

abstract class BaseViewModel : ViewModel(), BaseViewModel {
    override val throwableSingleLiveEvent: SingleLiveEvent<Throwable?> = SingleLiveEvent()

    override fun throwError(throwable: Throwable?) {
        throwableSingleLiveEvent.value = throwable
    }
}